controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -455,3 +487,399 @@ func (r *KafkaUserReconciler) getKafkaUserCredsFromSecret(
 
 	return string(username[:len(username)-1]), string(password[:len(password)-1]), nil
 }
+
+func (r *KafkaUserReconciler) getKafkaUserCertIDFromSecret(
+	ctx context.Context,
+	certRequest *v1beta1.CertificateRequest,
+) (string, error) {
+	kafkaUserCertSecret := &v1.Secret{}
+	kafkaUserCertSecretNamespacedName := types.NamespacedName{
+		Name:      certRequest.SecretName,
+		Namespace: certRequest.SecretNamespace,
+	}
+
+	err := r.Get(ctx, kafkaUserCertSecretNamespacedName, kafkaUserCertSecret)
+	if err != nil {
+		return "", err
+	}
+
+	certID := kafkaUserCertSecret.Data["id"]
```

Comment:
`certID := kafkaUserCertSecret.Data["id"]` -> `certID := kafkaUserCertSecret.StringData["id"]` 

I suggest you use stringData or Data, not both, cause it might be a little bit confusing, wdyt? Also, you can return `certID` instead of `string(certID)` from this function. Please fix it in all similar cases, if you agree.